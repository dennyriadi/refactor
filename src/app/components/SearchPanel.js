import React from 'react';
import PropTypes from 'prop-types';
import { compose, withState } from 'recompose';

const SearchPanel = props => {
  const {
    age,
    ageRange,
    gender,
    search,
    setAge,
    setAgeRange,
    setGender,
  } = props;

  const onChange = e => {
    const { name, value } = e.target;
    const funcMap = {
      gender: setGender,
      ageRange: setAgeRange,
      age: setAge,
    };

    return funcMap[name](value);
  };

  const onSubmit = e => {
    e.preventDefault();
    search({ age, ageRange, gender });
  };

  return (
    <div className="container-fluid p-3">
      <form onSubmit={onSubmit}>
        <div className="form-group">
          <label className="col-2">Gender:</label>
          <select name="gender" className="custom-select mr-2" onChange={onChange}>
            <option value="">All</option>
            <option value="male">Male</option>
            <option value="female">Female</option>
          </select>
        </div>
        <div className="form-group">
          <label className="col-2">Age: </label>
          <select name="ageRange" className="custom-select mr-2" onChange={onChange}>
            <option value="">Equal</option>
            <option value="lt">Younger than</option>
            <option value="gt">Older than</option>
          </select>
          <input
            name="age"
            type="number"
            min="0"
            className="form-control w-50 d-inline-block"
            placeholder="eg. 30, blank for all ages"
            onChange={onChange}
          />
        </div>
        <div className="form-group">
          <button type="submit" className="btn btn-primary">Search</button>
        </div>
      </form>
    </div>
  );
};

SearchPanel.defaultProps = {
  age: '',
  ageRange: '',
  gender: '',
};

SearchPanel.propTypes = {
  age: PropTypes.string,
  ageRange: PropTypes.string,
  gender: PropTypes.string,
  search: PropTypes.func.isRequired,
  setAge: PropTypes.func.isRequired,
  setAgeRange: PropTypes.func.isRequired,
  setGender: PropTypes.func.isRequired,
};

const enhance = compose(
  withState('gender', 'setGender', ''),
  withState('ageRange', 'setAgeRange', ''),
  withState('age', 'setAge', '')
);

export default enhance(SearchPanel);
