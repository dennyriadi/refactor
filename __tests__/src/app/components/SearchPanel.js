import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import SearchPanel from '../../../../src/app/components/SearchPanel';

describe('<SearchPanel />', () => {
  describe('render()', () => {
    test('should render correctly', () => {
      const wrapper = shallow(<SearchPanel search={jest.fn()} />);
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
