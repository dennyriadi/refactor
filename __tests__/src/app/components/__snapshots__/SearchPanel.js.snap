// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchPanel /> render() should render correctly 1`] = `
<withState(withState(SearchPanel))
  gender=""
  search={[Function]}
  setGender={[Function]}
/>
`;
